src/mbgl/text/placement.cpp

```diff
@@ -656,18 +661,17 @@ void Placement::placeBucket(const SymbolBucket& bucket,
         }
 
     } else {
-        for (const SymbolInstance& symbol : bucket.symbolInstances) {
-            placeSymbol(symbol);
+        auto endIt = bucket.symbolInstances.begin() + params.symbolInstanceEnd;
```

Comment:
do we check that `endIt <= bucket.symbolInstances.end()`?

Reply:
Nope, would an assert be what you're looking for?

Reply:
assert is fine (as long as there are no possible code paths that could violate it :smile: )